### PR TITLE
Remove deprecated component stability helpers

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan <bogdandrutu@gmail.com>

### CODE REVIEW

File: config/confighttp/confighttp.go [hunk 1]
```diff
@@ -183,11 +183,6 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
 	}, nil
 }
 
-// Deprecated: [v0.57.0] use ToClient.
```
Comment (mx-psi) [inline config/confighttp/confighttp.go:186]: This is missing a changelog entry

File: config/confighttp/confighttp.go [hunk 2]
```diff
@@ -183,11 +183,6 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
...
-func (hcs *HTTPClientSettings) ToClientWithHost(host component.Host, settings component.TelemetrySettings) (*http.Client, error) {
```
Comment (TylerHelmuth) [inline config/confighttp/confighttp.go:187]: Is this related to component stability helpers or should removing this deprecated function be a separate PR?

File: config/confighttp/confighttp.go (hunk 1 above)
Comment (bogdandrutu) [reply config/confighttp/confighttp.go]: Removed, will do a separate PR.

File: config/confighttp/confighttp.go (hunk 2 above)
Comment (bogdandrutu) [reply config/confighttp/confighttp.go]: Removed, will do a separate PR.

**Review by TylerHelmuth — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]
